File: clang/contador_desc_multihilo.c
#include <stdio.h>
#include <time.h>
#include <pthread.h>

#define N 1000000

void* countdown(void* arg) {
    int n = *(int*)arg;
    while (n > 0) {
        n--;
    }

    return NULL;
}


int main() {
    pthread_t t1, t2;
    int n1 = N / 2;
    int n2 = N / 2;

    clock_t inicio = clock();
    pthread_create(&t1, NULL, countdown, &n1);
    pthread_create(&t2, NULL, countdown, &n2);

    pthread_join(t1, NULL);
    pthread_join(t2, NULL);
    
    clock_t fin = clock();

    double t_ejecucion = (double)(fin - inicio) / CLOCKS_PER_SEC;

    printf("Tiempo de ejecucion (multihilo): %.6f segundos\r\n", t_ejecucion);

    return 0;
}

File: python/contador_desc_multihilo.py
import time
from threading import Thread

N = 100_000_000


def countdown(n: int):
    while n > 0:
        n -= 1
    

if __name__ == "__main__":
    inicio = time.perf_counter()
    t1 = Thread(target=countdown, args=(N // 2, ))
    t2 = Thread(target=countdown, args=(N // 2, ))
    t1.start()
    t2.start()
    t1.join()
    t2.join()
    fin = time.perf_counter()

    t_ejecucion = fin - inicio

    print(f"Tiempo de ejecución: {t_ejecucion:.6f} segundos")

